src/main/java/core/basesyntax/services/DataProcessorImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.services;
+
+import core.basesyntax.ActivityStrategy;
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.models.Product;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DataProcessorImpl implements DataProcessor {
+    private final ActivityStrategy activityStrategy;
+    private final FruitDao fruitDao;
+
+    public DataProcessorImpl(ActivityStrategy activityStrategy, FruitDao fruitDao) {
+        this.activityStrategy = activityStrategy;
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public Map<String, Integer> process() {
+        Map<String, Integer> reportsOfProducts = new HashMap<>();
+        for (Product product : fruitDao.getAll()) {
+            reportsOfProducts.putIfAbsent(product.getName(), 0);
+            Integer newQuantity = activityStrategy
+                    .getActivity(product.getType())
+                    .apply(reportsOfProducts.get(product.getName()), product.getQuantity());
+
+            reportsOfProducts.put(product.getName(), newQuantity);
+        }
+        return reportsOfProducts;
```

Comment:
Мапа імітує БД. База даних це структура, що є спільною для будь-якої частини програми. В вашому рішенні її імітує мапа PRODUCT_STORAGE. Давайте її використовувати.  
```suggestion
    public void process(List<FruitTransaction> transactions) {
        for (FruitTransaction transaction : transactions) {
           PRODUCT_STORAGE.putIfAbsent(product.getName(), 0); // це хардкод, подумайте про використання методу getOrDefault
            Integer newQuantity = activityStrategy
                    .getActivity(transaction.getType())
                    .apply(PRODUCT_STORAGE.get(transaction.getName()),transaction.getQuantity());

            PRODUCT_STORAGE.put(transaction.getName(), newQuantity);
        }
        return reportsOfProducts;
```